docs/parser_plugins.md

```diff
@@ -1 +1,176 @@
+The nesting of layers is done by referring back to `TextParser` via `Quantity.sub_parser: TextParser`.
+
+The main concern is how to leverage the additional freedom of a third format.
+Our foremost advice is to consider this tree an enriched representation of the source file and thus **follow the text file structure as faithfully as possible**.
+In its simplest form, this means
+
+1. following the order in which the data appears.
+2. capturing text blocks with a single `text_parser.py/Quantity`. Then you can further process its data via a deeper layer.
+
+Let us demonstrate the second point with FHI-aims output, which is typically partioned into blocks with headers and `|`-indented lines, e.g.
+
+```txt
+  End self-consistency iteration #     1       :  max(cpu_time)    wall_clock(cpu1)$
+  | Time for this iteration                    :        1.440 s           1.454 s$
+  | Charge density update                      :        0.500 s           0.504 s$
+  | Density mixing & preconditioning           :        0.000 s           0.000 s$
+  ...
+$
+```
+
+Here we indicate repetition as `...` and print the regex end-of-line symbol (`$`) for legibility.
+This block would be captured as
+
+```python
+
+```
+
+Note that the data extracted should be indicated between parentheses `()`, i.e. be a matching group in Python regex.
+To promote semantic extraction, we provide labelled patterns for common flags, e.g. `FHIAimsOutParser.re_float`.
+You can add a match group to a pattern via `capture(quantity)`.
+
+For blocks, the matching group should lie between the header and footer.
+Most times, devs prefer to approach the matching group content in a block as generic, non-descript text to focus on the stand-out characteristics of the block itself.
+You can use the `FHIAimsOutParser.re_non_greedy` flag here.
+Be careful: it should be superceded by a final regex pattern, else the matching continues up to the end-of-file.
+In our example, the block finishes with a blank line, denoted by the `FHIAimsOutParser.re_blank_line` flag. 
+
+A similar approach applies to processing individual tabular lines.
+Focus first on identifying the line as whole, which always end on `FHIAimsOutParser.re_eol`.
+Ensure that you toggle the `xxxx.Quantity.repeats` option to `True` or any non-zero number, so you match all lines.
+The actual parsing of the data then happens at a deeper layer.
+It is therefore not uncommon to encounter block parsing of up to three layers deep.
+
+Sometimes, it is easier to map the data by their column index.
+Here, you can again use `xxxx.Quantity.repeats` in the `sub_parser`, this time as a `dict` mapping the indices to their respective labels.
+This is just a convenient short-hand.
+Accessing the data works just as with any other nested structure.
+
+Following this logic through, you will end up with a continuation of a vertically connected tree.
```

Comment:
> Following this logic through, you will end up with a continuation of a vertically connected tree.

Following this logic ~~through~~ , you will end up with a continuation of a vertically connected tree.